src/Domain/Structure/DirectionId.hpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <iosfwd>
+
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/ElementId.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+template <size_t VolumeDim>
+struct DirectionId {
```

Comment:
Two thoughts: (1) DirectionId sounds like the ID for a direction, which made me wonder why we need an ID for it until I realized it stands for "Direction and neighbor ID". Related: (2) this same type is defined as MortarId in MortarHelpers.hpp. So maybe name it MortarId? It's not always used to identify DG mortars but IMO that's fine. If you don't like it maybe we can give it a thought on the Monday call.

Reply:
Just as a note in case I can't make the meeting for some unexpected reason, we used to have this data structure and called it `DirectionalElementId`.

I don't love the subcell code using `MortarId` since there are no mortars, really. Most of the places that would use it are actually indexing volume data. I worry that the subcell code is already too complicated with my best efforts to use clear names, so using a name like `MortarId` for indexing volume data, volume fluxes, neighbor RDMP data, etc. would make it extra confusing :(

Reply:
`DirectionalElementId` sounds good enough to me. We could alias `using MortarId = DirectionalElementId` in the `dg` namespace (not requesting this change). Why was the data structure removed (I don't remember it).

Reply:
Sounds good to both! My memory is that at some point we were only using it in a couple places and figured a `std::pair` was easier than having an extra class (turns out whatever group of us made that decision was wrong :P)

Reply:
I won't be able to make the meeting tomorrow.  I'll do whatever the consensus is, but  I'll suggest DirectionalId as that is shorter.   I presume you want the same name change for DirectionIdMap as well.